src/Process/PostDeploy/WarmUp/Urls.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Process\PostDeploy\WarmUp;
+
+use Magento\MagentoCloud\Config\Stage\PostDeployInterface;
+use Magento\MagentoCloud\Shell\ShellException;
+use Magento\MagentoCloud\Shell\ShellInterface;
+use Magento\MagentoCloud\Util\UrlManager;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Returns list of urls for warm up
+ */
+class Urls
+{
+    const ENTITY_CATEGORY = 'category';
+    const ENTITY_CMS_PAGE = 'cms-page';
+
+    /**
+     * @var PostDeployInterface
+     */
+    private $postDeploy;
+
+    /**
+     * @var UrlManager
+     */
+    private $urlManager;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Variable for caching base urls hosts from config table
+     *
+     * @var string
+     */
+    private $baseHosts;
+
+    /**
+     * @var ShellInterface
+     */
+    private $shell;
+
+    /**
+     * @param PostDeployInterface $postDeploy
+     * @param UrlManager $urlManager
+     * @param LoggerInterface $logger
+     * @param ShellInterface $shell
+     */
+    public function __construct(
+        PostDeployInterface $postDeploy,
+        UrlManager $urlManager,
+        LoggerInterface $logger,
+        ShellInterface $shell
+    ) {
+        $this->postDeploy = $postDeploy;
+        $this->urlManager = $urlManager;
+        $this->logger = $logger;
+        $this->shell = $shell;
+    }
+
+    /**
+     * Returns list of URLs which should be warm up.
+     *
```

Comment:
would be good to have a link to description of the url pattern. Or describe it here

Reply:
This description will be added to WARM_UP_PAGES variable.